packages/dids/src/resolver/universal-resolver.ts

```diff
@@ -121,16 +121,28 @@ export class UniversalResolver implements DidResolver, DidUrlDereferencer {
     }
 
     const cachedResolutionResult = await this.cache.get(parsedDid.uri);
-
     if (cachedResolutionResult) {
+      if (cachedResolutionResult?.isStaleInCache === true) {
+        // if the cache is stale, we would like to attempt refreshing it after returning the existing stale result
+        isomorphicNextTick(() => this.resolveAndCache(parsedDid.uri, resolver, options));
```

Comment:
haven't been in js land in a while, do we have a potential async issue here in that `resolveAndCache` is async but `isomorphicNextTick` is not, and we want to block until `resolveAndCache` is completed?

also, should `resolveAndCache` return value be set to `cachedResolutionResult`

so something like (probably not totally correct)

```suggestion
        cachedResolutionResult = await isomorphicNextTick(() => this.resolveAndCache(parsedDid.uri, resolver, options));
```

Reply:
In this particular case I didn't want to block while resolving + caching as the `dht` resolution can take up to 10 seconds. Talking with @csuwildcat we figure the chance of a stale `did` document being incorrect is probably very slim and is affected by general caching anyway so it's preferable to return a stale result than wait for a new one (which could also fail resolution intermittently).

So that was the purpose of introducing an `isomorphicNextTick` to allow js to trigger this after the function returns without having to block for it.

JS land is tricky with this kind of stuff but this *seems* to be the right way to do it.

Reply:
excellent